### ISSUE TITLE
Update cockroach demo --geo-partitioned-replicas docs to deemphasize / point to new MR patterns

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#10425]
Right now we have some [`cockroach demo` docs that use the old geo-partitioning ways](https://github.com/cockroachdb/cockroach/issues/63876).  That will still work in v21.1, but we should at least add a note saying "oh hey, here is [some new multi-region hotness you should use instead of this old geo-partitioning stuff](https://cockroachlabs.com/docs/v21.1/multiregion-overview.html)"

**Comment by rmloveland** [comment]
FYI @awoods187 and @ajstorm - I am planning to add a note alongside that deemphasizes this, rather than rip it out altogether, since AIUI this still exists in v21.1's `cockroach demo` 

(cc @ericharmeling since this touches many things you are and have worked on also)

**Comment by awoods187** [comment]
yeah i'm fine with that for now. I don't think we have to have documentation for every flag, particularly those ones we don't want people to use. But, I think we should keep this for one more release.
